src/app/components/form/input/FormInput.vue

```diff
@@ -1,47 +1,38 @@
 <template>
-  <div
-    class="flex flex-col gap-1"
-    :class="{
-      'form-input-success': success,
-      'form-input-warning': warning,
-      'form-input-error': value?.$error,
-    }"
-  >
+  <div class="flex flex-col gap-1">
     <div class="flex flex-col gap-2">
-      <div>
-        <label
-          class="inline-flex items-baseline gap-2 font-semibold"
-          :class="{
-            'form-input-success': success,
-            'form-input-warning': warning,
-            'form-input-error': value?.$error,
-          }"
-          :for="idLabelFull"
+      <label
+        class="inline-flex items-baseline gap-2"
+        :class="{
+          'text-(--semantic-critic-text)': value?.$error,
+        }"
+        :for="idLabelFull"
+      >
+        <TypographySubtitleMedium>{{ title }}</TypographySubtitleMedium>
+        <span
+          v-if="isRequired || isOptional"
+          class="text-xs font-medium text-gray-500 dark:text-gray-400"
         >
-          <TypographySubtitleMedium>{{ title }}</TypographySubtitleMedium>
-          <span
-            v-if="isRequired || isOptional"
-            class="text-xs font-medium text-gray-500 md:text-right dark:text-gray-400"
-          >
-            <span v-if="isRequired">
-              {{ t('required') }}
-            </span>
-            <span v-if="isOptional">
-              {{ t('optional') }}
-            </span>
+          <span v-if="isRequired">
+            {{ t('required') }}
           </span>
-        </label>
-      </div>
+          <span v-if="isOptional">
+            {{ t('optional') }}
+          </span>
+        </span>
+      </label>
       <div class="relative">
         <slot v-if="$slots.default" />
         <!-- TODO: support textarea, checkboxes and radio buttons natively -->
         <div
           v-else
-          class="flex items-center gap-2 rounded-lg border border-(--semantic-base-line) bg-(--semantic-base-input-field-fill)"
+          class="flex items-center gap-2 rounded-lg border bg-(--semantic-base-input-field-fill) has-[input[aria-invalid=true]]:border-(--critic-weak)"
         >
           <input
             :id="idLabelFull"
-            class="flex-1 border-none px-4 py-3 placeholder-(--semantic-base-text-secondary) outline-0"
+            class="peer flex-grow border-none px-4 py-3 placeholder-(--semantic-base-text-secondary) outline-0 focus:outline-none"
```

Comment:
I think we can just go with

```suggestion
            class="peer flex-grow border-none px-4 py-3 placeholder-(--semantic-base-text-secondary) outline-none"
```

and also keep `flex-1`. Btw, `flex-grow` [was deprecated and is removed](https://tailwindcss.com/docs/upgrade-guide#removed-deprecated-utilities).

Reply:
I made these changes except for `flex-shrink-0` which still needs to be changed.